File: RQ2_Prompt_Elements/OpenAI_Data/HumanEvalPython_output/scenario3/id_7T2000Test.py
# id_7_Test.java
from scenario3.id_7 import filter_by_substring

import pytest

class id_7_Test:
    """
    Test class of id_7.py.
    It contains ten unit test cases for the :func:`scenario3.id_7.filter_by_substring`.
    """


    def test_filter_by_substring_1(self):
        """
        Test case for the :func:`scenario3.id_7.filter_by_substring`.
        """
        assert filter_by_substring([], 'a') == []

    def test_filter_by_substring_2(self):
        """
        Test case for the :func:`scenario3.id_7.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'a') == ['abc', 'bacd', 'array']

    def test_filter_by_substring_3(self):
        """
        Test case for the :func:`scenario3.id_7.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'b') == ['abc', 'bacd']

    def test_filter_by_substring_4(self):
        """
        Test case for the :func:`scenario3.id_7.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'c') == ['abc', 'bacd', 'cde']

    def test_filter_by_substring_5(self):
        """
        Test case for the :func:`scenario3.id_7.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'd') == ['bacd']

    def test_filter_by_substring_6(self):
        """
        Test case for the :func:`scenario3.id_7.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'e') == ['cde']

    def test_filter_by_substring_7(self):
        """
        Test case for the :func:`scenario3.id_7.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'r') == ['array']

    def test_filter_by_substring_8(self):
        """
        Test case for the :func:`scenario3.id_7.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'y') == ['array']

    def test_filter_by_substring_9(self):
        """
        Test case for the :func:`scenario3.id_7.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'z') == []

    def test_filter_by_substring_10(self):
        """
        Test case for the :func:`scenario3.id_7.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'ab') == ['abc', 'bacd', 'array']

# id_8.py
from typing import List


def filter_by_substring(strings: List[str], substring: str) -> List[str]:
    """ Filter an input list of strings only for ones that contain given substring
    >>> filter_by_substring([], 'a')
    []
    >>> filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'a')
    ['abc', 'bacd', 'array']
    """

    pass

# id_8_Test.java
from scenario3.id_8 import filter_by_substring

import pytest

class id_8_Test:
    """
    Test class of id_8.py.
    It contains ten unit test cases for the :func:`scenario3.id_8.filter_by_substring`.
    """
		
    def test_filter_by_substring_1(self):
        """
        Test case for the :func:`scenario3.id_8.filter_by_substring`.
        """
        assert filter_by_substring([], 'a') == []

    def test_filter_by_substring_2(self):
        """
        Test case for the :func:`scenario3.id_8.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'a') == ['abc', 'bacd', 'array']

    def test_filter_by_substring_3(self):
        """
        Test case for the :func:`scenario3.id_8.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'b') == ['abc', 'bacd']

    def test_filter_by_substring_4(self):
        """
        Test case for the :func:`scenario3.id_8.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'c') == ['abc', 'bacd', 'cde']

    def test_filter_by_substring_5(self):
        """
        Test case for the :func:`scenario3.id_8.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'd') == ['bacd']

    def test_filter_by_substring_6(self):
        """
        Test case for the :func:`scenario3.id_8.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'e') == ['cde']

    def test_filter_by_substring_7(self):
        """
        Test case for the :func:`scenario3.id_8.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'r') == ['array']

    def test_filter_by_substring_8(self):
        """
        Test case for the :func:`scenario3.id_8.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'y') == ['array']

    def test_filter_by_substring_9(self):
        """
        Test case for the :func:`scenario3.id_8.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'z') == []

    def test_filter_by_substring_10(self):
        """
        Test case for the :func:`scenario3.id_8.filter_by_substring`.
        """
        assert filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'ab') == ['abc', 'bacd', 'array']

# id_9.py
from typing import List


def filter_by_substring(strings: List[str], substring: str) -> List[str]:
    """ Filter an input list of strings only for ones that contain given substring
    >>> filter_by_substring([], 'a')
    []
    >>> filter_by_substring(['abc', 'bacd', 'cde', 'array'], 'a')
    ['abc', 'bacd', 'array']
    """

    pass

# id_9_Test.java
from scenario3.id_9 import filter_by_substring

import pytest

class id_9_Test:
    """

File: RQ2_Prompt_Elements/GPT3.5_Data/HumanEvalJava_output/scenario3/id_7T2000Test.java
// FilterBySubstring.java
package scenario3;

import java.util.ArrayList;
import java.util.List;

class FilterBySubstring {

    /**
     * Filter an input list of strings only for ones that contain given substring
     * > filterBySubstring([], "a")
     * []
     * > filterBySubstring(["abc", "bacd", "cde", "array"], "a")
     * ["abc", "bacd", "array"]
     * > filterBySubstring([], "john")
     * []
     * > filterBySubstring(["xxx", "asd", "xxy", "john doe", "xxxAAA", "xxx"], "xxx")
     * ["xxx", "xxxAAA", "xxx"]
     * > filterBySubstring(["xxx", "asd", "aaaxxy", "john doe", "xxxAAA", "xxx"], "xx")
     * ["xxx", "aaaxxy", "xxxAAA", "xxx"]
     * > filterBySubstring(["grunt", "trumpet", "prune", "gruesome"], "run")
     * ["grunt", "prune"]
     */
    public static List<String> filterBySubstring(List<String> strings, String substring) {
        List<String> filteredStrings = new ArrayList<>();
        for (String str : strings) {
            if (str.contains(substring)) {
                filteredStrings.add(str);
            }
        }
        return filteredStrings;
    }
}

// FilterBySubstringTest.java
package scenario3;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link FilterBySubstring}.
* It contains ten unit test cases for the {@link FilterBySubstring#filterBySubstring(List, String)} method.
*/
class FilterBySubstringTest {

    @Test
    void testEmptyList() {
        List<String> input = Collections.emptyList();
        List<String> expectedOutput = Collections.emptyList();
        assertEquals(expectedOutput, FilterBySubstring.filterBySubstring(input, "a"));
    }

    @Test
    void testNoMatch() {
        List<String> input = Arrays.asList("abc", "def", "ghi");
        List<String> expectedOutput = Collections.emptyList();
        assertEquals(expectedOutput, FilterBySubstring.filterBySubstring(input, "j"));
    }

    @Test
    void testSingleMatch() {
        List<String> input = Arrays.asList("abc", "def", "ghi");
        List<String> expectedOutput = Arrays.asList("abc");
        assertEquals(expectedOutput, FilterBySubstring.filterBySubstring(input, "a"));
    }

    @Test
    void testMultipleMatches() {
        List<String> input = Arrays.asList("abc", "bacd", "cde", "array");
        List<String> expectedOutput = Arrays.asList("abc", "bacd", "array");
        assertEquals(expectedOutput, FilterBySubstring.filterBySubstring(input, "a"));
    }

    @Test
    void testMatchAtBeginning() {
        List<String> input = Arrays.asList("xxx", "asd", "xxy", "john doe", "xxxAAA", "xxx");
        List<String> expectedOutput = Arrays.asList("xxx", "xxxAAA", "xxx");
        assertEquals(expectedOutput, FilterBySubstring.filterBySubstring(input, "xxx"));
    }

    @Test
    void testMatchAtEnd() {
        List<String> input = Arrays.asList("xxx", "asd", "aaaxxy", "john doe", "xxxAAA", "xxx");
        List<String> expectedOutput = Arrays.asList("xxx", "aaaxxy", "xxxAAA", "xxx");
        assertEquals(expectedOutput, FilterBySubstring.filterBySubstring(input, "xx"));
    }

    @Test
    void testMatchInMiddle() {
        List<String> input = Arrays.asList("grunt", "trumpet", "prune", "gruesome");
        List<String> expectedOutput = Arrays.asList("grunt", "prune");
        assertEquals(expectedOutput, FilterBySubstring.filterBySubstring(input, "run"));
    }

    @Test
    void testEmptySubstring() {
        List<String> input = Arrays.asList("abc", "def", "ghi");
        List<String> expectedOutput = Collections.emptyList();
        assertEquals(expectedOutput, FilterBySubstring.filterBySubstring(input, ""));
    }

    @Test
    void testNullInput() {
        List<String> input = null;
        assertThrows(NullPointerException.class, () -> FilterBySubstring.filterBySubstring(input, "a"));
    }

    @Test
    void testNullSubstring() {
        List<String> input = Arrays.asList("abc", "def", "ghi");
        assertThrows(NullPointerException.class, () -> FilterBySubstring.filterBySubstring(input, null));
    }
}